v2/internal/resolver/resource_hierarchy.go

```diff
@@ -202,13 +219,9 @@ func (h ResourceHierarchy) fullyQualifiedARMIDImpl(subscriptionID string, origin
 		if err != nil {
```

Comment:
If you take the `checkARMIDHasSubscription` suggestion below, also consider renaming this `armID` variable to `ownerARMID` so the check reads: `checkARMIDHasSubscription(subscriptionID, ownerARMID)`